src/util/panic.js

```diff
@@ -57,9 +73,17 @@ function memory_monitor() {
         const align = h.step - (increase % h.step);
         h.next += increase + align;
     }
+    if (mp && mp.next && mp.next <= current) {
```

Comment:
Perhaps this block should come before the heapdump one, because we know heapdump has crashed the process before, so it just better be last so we get all the info we can.